README.md

```diff
@@ -0,0 +1,129 @@
+### To set up machine to build OMS
+
+There are two ways to build OMS.
+
+1. As an RPM or DEB package that can be installed on the local system,
+assuming all dependencies are met (most easily met if a shell bundle
+was previously installed to "seed" the system), or
+2. As a shell bundle.
+
+Building a shell bundle is a superset of of setting up a system to
+build a local RPM, so we will cover that first.
+
+#### To set up system to build a native package
+
+Note that it's very nice to be able to use the [updatedns]
+(https://github.com/jeffaco/msft-updatedns) project to
+use host names rather than IP numbers. On CentOS systems, this requires
+the bind-utils package (updatedns requires the 'dig' program). The
+bind-utils package isn't otherwise necessary.
+
+- On CentOS 7.x
+```
+ sudo yum install git bind-utils ruby bison gcc-c++ rpm-devel pam-devel openssl-devel rpm-build
+```
+- On Ubuntu 14.04
+```
+ sudo apt-get install git pkg-config make ruby bison g++ rpm librpm-dev libpam0g-dev libssl-dev
+```
+
+- Notes on other platforms
+
+ When building a machine for ULINUX builds (such as CentOS 5.x), we suggest
+ using the O/S distribution CD to install the packages. It's not as easy,
+ but that's the only way to guarentee that packages aren't updated such that
+ generated binaries are not backwards compatible. (See notes on building a
+ shell bundle, elsewhere in this document.)
+
+ Also note that since you won't use 'yum', you must also handle the dependent
+ packages manually (keep adding lines to the 'rpm install' command line until
+ all dependencies are satisfied).
+
+ Similar methods would be utilized if building a Redhat system that is not
+ registered for use for up2date.
+
+##### Notes on sudoers configuragion
+
+Two changes should be made to your sudoers configuration for omsagent to
+build properly:
+
+1. Configure sudoers to not require a TTY.
+
+ Some platforms require a TTY be default, and this can be problematic for
+ background builds. If you have a line like:
+
+ ```Defaults    requiretty```
+
+ then comment it out (like this):
```

Comment:
Could mention the use of visudo which prevents generating an invalid sudoers file and causing havoc.